src/pudl/transform/ferc1.py

```diff
@@ -761,56 +761,48 @@ def reconcile_table_calculations(
 
     Args:
         df: processed table.
-        tbl_meta: processed table xbrl metadata.
+        calculation_components: processed calculation component metadata.
         xbrl_factoid_name: column name of the XBRL factoid in the processed table.
         table_name: name of the PUDL table.
         params: :class:`ReconcileTableCalculations` parameters.
     """
     # If we don't have this value, we aren't doing any calculation checking:
     if params.column_to_check is None:
         return df
-
-    # skip the calculations that have any components from other tables.
-    # this could be removed/moved to when we deal with inter-table calcs.
-    inter_table_calculated_values = list(
-        tbl_meta[~tbl_meta.intra_table_calc_flag].xbrl_factoid.unique()
+    # we only want to check calucations that are fully within this table
+    intra_tbl_calcs = calculation_components[
+        calculation_components.intra_table_calc_flag
+        & calculation_components.xbrl_factoid.notnull()  # no nulls bc we have all parents
+    ]
+    # !!! Add dimensions into the calculation components!!!
```

Comment:
i ended up added the implied dimensions into the calculation component table in here because it made the calculation checking simpler and more in line with how we are checking calcs over in output land. I think this really belongs over in `process_xbrl_metadata_calculations`.

Reply:
yes the answer is yes

Reply:
Agree!

Reply:
hm we have a problem. adding in the dimensions requires having the data. bc the implied dimensions are gleaned from the processed data. typically we've done all of the metadata and calculation processing before and fully independent from the data processing.

I think because of this interdependency, I'd like to keep this as is.

Reply:
I don't know if this would make sense, but in the context of the assets being written into the database, we could have per-table calculation component tables that do depend on the data (taking the data tables as inputs).

But maybe this is duplicative with the all-tables calculation components table, which will have a more complete knowledge of all the dimensional values that are observed?